src/ansys_sphinx_theme/search/fuse_search.py

```diff
@@ -204,7 +204,26 @@ def create_search_index(app, exception):
 
     search_index_list = []
 
-    for document in app.env.found_docs:
+    static_search_options = app.config.html_theme_options.get("static_search", {})
+    excluded_docs = static_search_options.get("files_to_exclude", [])
+
+    for exclude_doc in excluded_docs:
+        is_folder = exclude_doc.endswith("/")
+
+        # Remove trailing slash for consistency in folder exclusion
+        if is_folder:
+            exclude_doc = exclude_doc.rstrip("/")
+
+        # Filter out documents based on whether exclude_doc is a folder or a file:
+        # - If a folder, exclude all documents starting with the folder name.
+        # - If a file, exclude only the exact file name.
+        included_docs = [
+            doc
+            for doc in app.env.found_docs
+            if not (is_folder and doc.startswith(exclude_doc)) and doc != exclude_doc
+        ]
+
+    for document in included_docs:
```

Comment:
Shouldn't you define a default value for `included_docs` ? If `excluded_docs` is empty, you'll never define that variables.

Reply:
added default value, thanks Sebastian